gh_actions_exporter/tokenGenerator.py

```diff
@@ -0,0 +1,25 @@
+import time
+
+from githubkit import AppInstallationAuthStrategy, GitHub
+
+from gh_actions_exporter.config import Settings
+
+
+class TokenGenerator:
+    def __init__(self, settings: Settings) -> None:
+        self.settings = settings
+        self.github_app_private_key: str = (
+            settings.github_app_private_key.get_secret_value()
+        )
+        self.github_app_id: int = int(settings.github_app_id)
+        self.github_app_installation_id: int = int(settings.github_app_installation_id)
+
+    def generate_token(self) -> GitHub:
+        github = GitHub(
+            AppInstallationAuthStrategy(
+                self.github_app_id,
+                self.github_app_private_key,
+                self.github_app_installation_id,
+            )
+        )
+        return github
```

Comment:
* This whole class should actually be a function defined as a fastapi dependency. Similar to what you did with cost.
* The name does not fit its function. You talk about tokengenrator but you return a client.

Reply:
Done! Tell me if it's better now

Reply:
Are we missing a git push?